feat: added logs quick filters product docs (#1003)
SigNoz/signoz.io: data/docs/userguide/logs_quick_filters.mdx

> ---

Reviewer: Can you please put this file inside logs-management folder under a sub folder called `features`

date: 2024-12-02
title: Logs Quick Filters

> id: logs_quick_filters

Reviewer: ```suggestion
id: logs-quick-filters
```

Please change the file name to `logs-quick-filters.mdx` as well.

---

The Logs Explorer comes with quick filters which allows filtering of logs based on key attributes making it easier to analyze and debug applications.

> Here's a detailed explaination of the available filters and how to use them -

Reviewer: ```suggestion
Below is a detailed explanation of the available filters and instructions on how to use them.
```

<figure data-zoomable align="center">
  <img
    src="/img/docs/logs-management/ui/logs-quick-filters.webp"
    alt="Quick Filters in Logs Explorer"
  />
  <figcaption>
    <i> Quick Filters in Logs Explorer </i>
  </figcaption>
</figure>

> ### Available Filters

Reviewer: ```suggestion
## Available Filters
```

> 

Reviewer: There are multiple filtering options available, with each filter mapped to a specific [OpenTelemetry (OTEL) attribute](https://opentelemetry.io/docs/specs/otel/common/#attribute). Here’s a list of the available filters:

1. **Severity Text**:
   - **Description**: Filter logs based on their severity text.
   - **Usage**: Useful for quickly filtering in FATAL logs when some incident occurs or filter in DEBUG logs when debugging some issue.

>    - **OTEL Attribute**: Maps to `severity_text` opentelemetry attribute.

Reviewer: ```suggestion
   - **OTel Attribute**: Maps to [`severity_text`](https://opentelemetry.io/docs/specs/otel/logs/data-model/#field-severitytext) opentelemetry attribute.
```
https://opentelemetry.io/docs/contributing/style-guide/#opentelemetryio-word-list

Please confirm if this hyperlink is correct - https://opentelemetry.io/docs/specs/otel/logs/data-model/#field-severitytext

2. **Environemnt**:

   - **Description**: Filter logs based on the source environment.
   - **Usage**: Useful for filtering in/out logs coming from different environments. Example - Production / Staging / Development etc.

>    - **OTEL Attribute**: Maps to `deployment.environment` opentelemetry resource attribute.

Reviewer: ```suggestion
   - **OTel Attribute**: Maps to [`deployment.environment`](https://opentelemetry.io/docs/specs/semconv/attributes-registry/deployment/#deployment-attributes) resource attribute.
```

3. **Service Name**:

   - **Description**: Filter logs by the name of the service that generated them.
   - **Usage**: Select from a list of service names to isolate logs from a specific service. This can be helpful in microservices architectures where multiple services generate logs.

>    - **OTEL Attribute**: Maps to `service.name` opentelemetry resource attribute.

Reviewer: ```suggestion
   - **OTel Attribute**: Maps to [`service.name`](https://opentelemetry.io/docs/specs/semconv/attributes-registry/service/#service-attributes) resource attribute.
```

4. **Hostname**:

   - **Description**: Filter logs based on the hostname of the machine that generated the logs.
   - **Usage**: This filter allows you to isolate logs from specific hosts, which is particularly useful in environments with multiple servers or instances. For example, you can filter logs to view only those generated by a particular server during a specific time frame.

>    - **OTEL Attribute**: Maps to `host.name` opentelemetry resource attribute.

Reviewer: ```suggestion
   - **OTel Attribute**: Maps to [`host.name`](https://opentelemetry.io/docs/specs/semconv/attributes-registry/host/#host-attributes) resource attribute.
```

5. **K8s Pod Name**:

   - **Description**: Filter logs based on the name of the Kubernetes pod that generated them.
   - **Usage**: This filter allows you to isolate logs from specific pods, which is particularly useful in environments with multiple pods running the same service.

>    - **OTEL Attribute**: Maps to `k8s.pod.name` opentelemetry resource attribute.

Reviewer: ```suggestion
   - **OTel Attribute**: Maps to [`k8s.pod.name`](https://opentelemetry.io/docs/specs/semconv/attributes-registry/k8s/#kubernetes-attributes) resource attribute.
```

6. **K8s Node Name**:

   - **Description**: Filter logs based on the name of the Kubernetes node that generated them.
   - **Usage**: This filter helps in isolating logs from specific nodes, which can be useful for debugging issues related to particular nodes in a cluster.

>    - **OTEL Attribute**: Maps to `k8s.node.name` opentelemetry resource attribute.

Reviewer: ```suggestion
   - **OTel Attribute**: Maps to [`k8s.node.name`](https://opentelemetry.io/docs/specs/semconv/attributes-registry/k8s/#kubernetes-attributes) resource attribute.
```

7. **K8s Cluster Name**:
   - **Description**: Filter logs based on the name of the Kubernetes cluster that generated them.
   - **Usage**: This filter allows you to isolate logs from specific clusters, which is particularly useful in environments with multiple clusters running different services.

>    - **OTEL Attribute**: Maps to `k8s.cluster.name` opentelemetry resource attribute.

Reviewer: ```suggestion
   - **OTel Attribute**: Maps to [`k8s.cluster.name`](https://opentelemetry.io/docs/specs/semconv/attributes-registry/k8s/#kubernetes-attributes) resource attribute.
```

---
date: 2024-12-02
title: Logs Quick Filters
id: logs_quick_filters
---

The Logs Explorer comes with quick filters which allows filtering of logs based on key attributes making it easier to analyze and debug applications.
Here's a detailed explaination of the available filters and how to use them -

<figure data-zoomable align="center">
  <img
    src="/img/docs/logs-management/ui/logs-quick-filters.webp"
    alt="Quick Filters in Logs Explorer"
  />
  <figcaption>
    <i> Quick Filters in Logs Explorer </i>
  </figcaption>
</figure>

### Available Filters

> 1. **Severity Text**:

Reviewer: ```suggestion
**Severity Text**:
```
```suggestion
### Severity Text
```

   - **Description**: Filter logs based on their severity text.
   - **Usage**: Useful for quickly filtering in FATAL logs when some incident occurs or filter in DEBUG logs when debugging some issue.
   - **OTEL Attribute**: Maps to `severity_text` opentelemetry attribute.

> 2. **Environemnt**:

Reviewer: ```suggestion
### Environment
```

   - **Description**: Filter logs based on the source environment.
   - **Usage**: Useful for filtering in/out logs coming from different environments. Example - Production / Staging / Development etc.
   - **OTEL Attribute**: Maps to `deployment.environment` opentelemetry resource attribute.

> 3. **Service Name**:

Reviewer: ```suggestion
### Service Name
```

   - **Description**: Filter logs by the name of the service that generated them.
   - **Usage**: Select from a list of service names to isolate logs from a specific service. This can be helpful in microservices architectures where multiple services generate logs.
   - **OTEL Attribute**: Maps to `service.name` opentelemetry resource attribute.

> 4. **Hostname**:

Reviewer: ```suggestion
### Hostname
```

   - **Description**: Filter logs based on the hostname of the machine that generated the logs.
   - **Usage**: This filter allows you to isolate logs from specific hosts, which is particularly useful in environments with multiple servers or instances. For example, you can filter logs to view only those generated by a particular server during a specific time frame.
   - **OTEL Attribute**: Maps to `host.name` opentelemetry resource attribute.

> 5. **K8s Pod Name**:

Reviewer: ```suggestion
### K8s Pod Name
```

   - **Description**: Filter logs based on the name of the Kubernetes pod that generated them.
   - **Usage**: This filter allows you to isolate logs from specific pods, which is particularly useful in environments with multiple pods running the same service.
   - **OTEL Attribute**: Maps to `k8s.pod.name` opentelemetry resource attribute.

> 6. **K8s Node Name**:

Reviewer: ```suggestion
### K8s Node Name
```

   - **Description**: Filter logs based on the name of the Kubernetes node that generated them.
   - **Usage**: This filter helps in isolating logs from specific nodes, which can be useful for debugging issues related to particular nodes in a cluster.
   - **OTEL Attribute**: Maps to `k8s.node.name` opentelemetry resource attribute.

> 7. **K8s Cluster Name**:

Reviewer: ```suggestion
### K8s Cluster Name
```

   - **Description**: Filter logs based on the name of the Kubernetes cluster that generated them.
   - **Usage**: This filter allows you to isolate logs from specific clusters, which is particularly useful in environments with multiple clusters running different services.
   - **OTEL Attribute**: Maps to `k8s.cluster.name` opentelemetry resource attribute.

<Admonition>

>   For mapping attributes from log body to the above attributes then use{' '}

Reviewer 2: For mapping attributes from log body to the above attributes ~~then~~ use

remove `then`

Author: done @nityanandagohain

  <a href="https://signoz.io/docs/logs-pipelines/introduction/">log pipelines</a>
</Admonition>

> ### How to use quick filters

Reviewer: ```suggestion
## How to use quick filters
```
